src/ray/rpc/server_call.h

```diff
@@ -194,21 +203,46 @@ class ServerCallImpl : public ServerCall {
   void SetState(const ServerCallState &new_state) override { state_ = new_state; }
 
   void HandleRequest() override {
+    bool auth_success = true;
+    if (::RayConfig::instance().enable_cluster_auth()) {
+      if constexpr (EnableAuth == AuthType::LAZY_AUTH) {
+        RAY_CHECK(!cluster_id_.IsNil()) << "Expected cluster ID in server call!";
+        auto &metadata = context_.client_metadata();
+        if (auto it = metadata.find(kClusterIdKey);
+            it != metadata.end() && it->second != cluster_id_.Hex()) {
+          RAY_LOG(DEBUG) << "Wrong cluster ID token in request! Expected: "
+                         << cluster_id_.Hex() << ", but got: "
+                         << (it == metadata.end() ? "No token!" : it->second);
+          auth_success = false;
+        }
+      } else {
+        if (!cluster_id_.IsNil()) {
+          RAY_LOG_EVERY_MS(WARNING, 5000)
```

Comment:
Should this be RAY_CHECK as well?

Reply:
This one is for the NO_AUTH case

Reply:
(so wrong cluster ID is fine)

Reply:
I guess technically if we don't need it and we have it stored locally, it's not a big deal. I did change the actual warning log to log whether the client has metadata, though.

Reply:
Actually left it as is. There's not much value to checking the client one, and I think periodically logging that the object has a cluster ID that it doesn't need is fine.